src/components/use-oauth2.test.ts

```diff
@@ -169,7 +169,7 @@ describe('useOAuth2', () => {
 					REDIRECT_URI,
 					generatedState as string
 				),
-				{ method: 'POST' },
+				{ method: 'POST', headers: {} },
```

Comment:
It would be nice to send some actual headers here and verify from server side that they are being sent :)

Reply:
I've added a bearer token (just taken the sample token from https://jwt.io/) for the comparison of the result accordingly.